### ISSUE TITLE
How do I get the HTTP status code in the error case?

### ISSUE THREAD

**Issue by njlr** [fable-compiler/fable-fetch#10]
It looks like the response gets turned into a string for all response codes other than `2xx`: 

https://github.com/fable-compiler/fable-fetch/blob/b79a153b59f806771508fa36321221e5dc338269/src/Fetch.fs#L379

However, I need to handle HTTP codes differently in my app. 

How can I get the HTTP code for error cases (e.g. `401`)?

**Comment by MangelMaxime** [comment]
Hello with the current implementation you can't access it, it needs to be rewritten.

For example, in [Thoth.Fetch](https://thoth-org.github.io//Thoth.Fetch/) we use a custom implementation to preserve the error information. 

Right now, I guess there is 3 possibility:

- Someone steps in to improve Fable.Fetch
- Write your own implementation inside of your project
- Use a library which provides support for it like Thoth.Fetch, perhaps there are others I don't know. If you are using Thoth.Json I can only encourage you to use Thoth.Fetch :) Just for clarity, I am the maintainer of it.

**Comment by njlr** [comment]
I started using [SimpleHttp](https://github.com/Zaid-Ajaj/Fable.SimpleHttp/), but only because that's the one I stumbled upon first. 

I will leave this open since having the error code seems like an important addition to `Fable.Fetch`.

**Comment by forki** [comment]
@MangelMaxime  we should provide a method that just runs https://github.com/fable-compiler/fable-fetch/blob/master/src/Fetch.fs#L375 without throwing if result is not ok

**Comment by MangelMaxime** [comment]
@forki Indeed, it should be enough I think for `fetch` method but not for `tryFetch`. 

I guess for `tryFetch` we will do something like that:

```fs
let tryFetch (url: string) (init: RequestProperties list) : JS.Promise<Result<Response, Response>> =
    GlobalFetch.fetch(RequestInfo.Url url, requestProps init)
    |> Promise.map (fun response ->
        if response.Ok
        then Ok response
        else Error response)
```

I didn't test the code so it is possible that it doesn't compile.
